packages/core/src/styles/tokens/tokens.stories.ts

```diff
@@ -60,20 +64,58 @@ function getTokenTable(key: string) {
     </table>`;
 }
 
-function getTokens(val: string) {
+let currentTheme = getThemeStyleMap(':root');
+const priorTheme = ':root';
+
+function getTokens(val: string, globals: any) {
+  if (priorTheme !== globals.theme) {
+    currentTheme = getThemeStyleMap(globals.theme ? `[cds-theme~="${globals.theme}"]` : ':root');
+  }
+
   return Object.entries(tokenData)
     .filter(([key]) => key.includes(val))
     .map((token: [string, any]) => {
-      return createToken({
-        name: tokenToCssProp(token[0]),
-        alias: token[1].alias ? tokenToCssProp(token[1].alias) : '',
-        value: token[1].value,
-      });
+      return createToken(
+        {
+          name: tokenToCssProp(token[0]),
+          alias: token[1].alias ? tokenToCssProp(token[1].alias) : '',
+          value: token[1].value,
+        },
+        globals,
+        currentTheme
+      );
     });
 }
 
-function createToken(token: Token) {
-  token.formattedValue = html`<span cds-text="code">${token.value}</span>`;
+function getComputedTokenValue(token: Token, globals: any) {
+  const el = document.createElement('div');
+  el.setAttribute('cds-theme', globals.theme);
+  document.body.appendChild(el);
+  const value = getComputedStyle(el).getPropertyValue(token.name);
+  document.body.removeChild(el);
+  return value;
+}
+
+function getThemeStyleMap(themeSelector: string) {
+  const sheet = Array.from(document.styleSheets)
+    .filter(sheet => sheet.href === null || sheet.href.startsWith(window.location.origin))
+    .filter(sheet => Array.from(sheet.cssRules).find((rule: any) => rule.selectorText === themeSelector))[0];
+
+  return sheet.cssRules[0].cssText
+    .replace(`${themeSelector} {`, '')
+    .replace('}', '')
+    .split(';')
+    .filter((r: any) => r.length > 1)
+    .reduce((prev: any, next: string) => {
+      const rule = next.split(':');
+      return { ...prev, [rule[0].trim()]: rule[1].trim() };
+    }, {});
+}
+
+function createToken(token: Token, globals: any, currentTheme: any) {
+  token.computedValue = getComputedTokenValue(token, globals);
+  token.formattedValue = html`<span cds-text="code">${token.computedValue}</span>`;
+  token.alias = token.alias ? currentTheme[token.name] : undefined;
```

Comment:
This addition allows the token to be read from the :root theme active in the browser. This enables the text to render the current value of the token when the theme is switched in storybook.